code/modules/mob/mob.dm

```diff
@@ -397,6 +397,16 @@
 /mob/proc/show_inv(mob/user)
 	return
 
+/mob/verb/giveitem(atom/A as mob in range(1))
+	set name = "Give"
+	set category = "IC"
+	if(!iscarbon(src))
+		to_chat(src, "<span class='warning'>You can't give items!</span>")
+		return
+	if(A && A != src && get_dist(src, A) < 2)
+		var/mob/living/carbon/C = src
+		C.give()
```

Comment:
```suggestion
/mob/living/carbon/verb/giveitem(var/mob/living/carbon/target in (view(1) - usr)))
	set name = "Give"
	set category = "IC"
	if(!iscarbon(target) || target.incapacitated() || !target.client)
		return
	give()
```
Think you could get a try at making this a pointed action? I can help out with that if needed.
The current implementation also kind of offers it to anyone, not to specific people.

Reply:
thats the point, youre holding it outwards so anyone can grab it.

Reply:
The way the Bay proc does it is targeting one person, which makes much more sense to me.
Else you'd have to give the take action to anyone that approaches as well if you want to implement snatching.